tests/tests/device.rs

```diff
@@ -550,6 +550,37 @@ static DEVICE_DROP_THEN_LOST: GpuTestConfiguration = GpuTestConfiguration::new()
         );
     });
 
+#[gpu_test]
+static DEVICE_INVALID_THEN_SET_LOST_CALLBACK: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(TestParameters::default().expect_fail(FailureCase::webgl2()))
+    .run_sync(|ctx| {
+        // This test checks that when the device is invalid, a subsequent call
+        // to set the device lost callback will immediately call the callback.
+        // Invalidating the device is done via a testing-only method. Fails on
+        // webgl because webgl doesn't implement make_invalid.
+
+        // Make the device invalid.
+        ctx.device.make_invalid();
+
+        let was_called = std::sync::Arc::<std::sync::atomic::AtomicBool>::new(false.into());
```

Comment:
```suggestion
        let was_called = std::sync::atomic::AtomicBool::new(false);
```

Reply:
Sorry, can't get this to work. I need to make a clone of the bool for the closure, otherwise it gets moved into the closure and then I can't check it afterwards. It's possible I'm missing something about how to do this cleanly in Rust, but this change (alone) does not work.

Reply:
I changed the `AtomicBool`s to be `static` items with 572efa1 and e33c4c3. They're `fixup!` commits, so you can squash 'em down with `git rebase --autosquash -i trunk` after you've had a gander. 🙂